### DISCUSSION TITLE
Drag and drop files here or 'Choose files'

### DISCUSSION THREAD

**Discussion by andymoss461** [UKHomeOffice/design-system#450] [General]
This component has been tested with seven internal users. Not all users directly interacted with the component (due to difficulties sharing the prototype remotely / giving control via Teams)

**Findings**
- The component is recognisable as a place to upload files
- Users tend not to use the drag and drop functionality but instead click 'Choose files'
- Users may upload files one at a time, even if they appear in the same folder

- The component offers 'drag and drop' on mobile, which doesn't make sense to users
- On mobile, users may wish to take a photo at the point of uploading. Android phones do offer this option when you press 'Choose files', but this isn't clear to users.

- Users are reassured by the image playback / upload confirmation, and may use it to check that they have uploaded the correct image

**Comments**
- Some image playback didn't appear correctly in the prototype (either failing to upload or not remembering what's been uploaded when the user goes back to the page)
- We haven't tested with files other than .jpg
- The component is being tested alongside a checklist for staff to check that they have uploaded the required evidence

**Next steps**
- @tomyems thinking about mobile option for testing
- Testing both desktop and mobile versions as we continue work on a new internal form

![Screenshot 2022-08-24 at 10 53 46](https://user-images.githubusercontent.com/97909786/186415585-0b8f2c09-1b9c-4052-943d-d7069b590a29.png)

**Comment by tomyems** [comment]
Ah amazing, thanks Andy, this is an implementation of the [Ministry of Justice multi file upload](https://design-patterns.service.justice.gov.uk/components/multi-file-upload/)

**Comment by isobelseacombe** [comment]
I've had a similar experience when testing this drag and drop component - users understand what it does but just click the choose files button instead. If this is the case, is there really a need for the drag and drop option? In two products I've worked on we ended up just going for the standard native file browser https://design-system.service.gov.uk/components/file-upload/

**Reply by tomyems** [reply]
That's consistent with what we've found (hence creating a discussion to record what we've found). How did you show users what they'd uploaded, did you do it on the same screen or on check your answers? We've got something like this:

![Screenshot 2022-08-24 at 16 04 56](https://user-images.githubusercontent.com/93599300/186453776-0ea74258-3976-4858-9cad-b6971527ef2d.png)

**Comment by isobelseacombe** [comment]
<img width="1018" alt="Screenshot 2022-08-24 at 16 11 09" src="https://user-images.githubusercontent.com/43995752/186455457-d2340d88-fe06-4e8a-827c-30291d9a49c0.png">
<img width="1019" alt="Screenshot 2022-08-24 at 16 11 18" src="https://user-images.githubusercontent.com/43995752/186455465-835c2519-8908-41a6-af18-3351bcd44bb2.png">

In this scenario we were asking the user to upload a photo and we found there was a need to get immediate visual feedback to reassure them that they had chosen the correct image

**Reply by isobelseacombe** [reply]
So yeah, same as you I think with showing the files straight away, rather than waiting until 'check answers'
